googletest/include/gtest/gtest.h

```diff
@@ -2116,9 +2116,11 @@ GTEST_API_ AssertionResult DoubleLE(const char* expr1, const char* expr2,
 // of the dummy variable name, thus allowing multiple SCOPED_TRACE()s
 // to appear in the same block - as long as they are on different
 // lines.
-#define SCOPED_TRACE(message) \
-  ::testing::internal::ScopedTrace GTEST_CONCAT_TOKEN_(gtest_trace_, __LINE__)(\
-    __FILE__, __LINE__, ::testing::Message() << (message))
+#define SCOPED_TRACE_(message, file, line) \
```

Comment:
http://stackoverflow.com/questions/16683146/can-macros-be-overloaded-by-number-of-arguments

On Tue, Nov 24, 2015 at 11:30 AM, ⭐caitp⭐ <notifications@github.com> wrote:

> In googletest/include/gtest/gtest.h
> <https://github.com/google/googletest/pull/620#discussion_r45758643>:
>
> > @@ -2116,9 +2116,11 @@ GTEST_API_ AssertionResult DoubleLE(const char* expr1, const char* expr2,
> >  // of the dummy variable name, thus allowing multiple SCOPED_TRACE()s
> >  // to appear in the same block - as long as they are on different
> >  // lines.
> > -#define SCOPED_TRACE(message) \
> > -  ::testing::internal::ScopedTrace GTEST_CONCAT_TOKEN_(gtest_trace_, __LINE__)(\
> > -    __FILE__, __LINE__, ::testing::Message() << (message))
> > +#define SCOPED_TRACE_(message, file, line) \
>
> as far as I'm aware, the only way to do this would still require defining
> multiple macro names (eg selecting the macro impl based on number of
> arguments). I don't know if that's any better
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/googletest/pull/620/files#r45758643>.
>